prowler/providers/aws/aws_provider.py

```diff
@@ -881,7 +888,174 @@ def get_checks_to_execute_by_audit_resources(self) -> set[str]:
+
+            >>> AwsProvider.test_connection(profile="test")
+            AWSCallerIdentity(user_id='AROAAAAAAAAAAAAAAAAAA:test-user', account='111122223333', arn=ARN(arn='arn:aws:sts::111122223333:user/test-user', partition='aws', service='sts', region=None, account_id='111122223333', resource='test-user', resource_type='user'), region='us-east-1')
+        """
+        try:
+            # Create the default session if no session is given
+            session = AwsProvider.setup_session(mfa_enabled, profile)
+
+            # Test Connection using the IAM Role
+            if role_arn:
+                session_duration = validate_session_duration(session_duration)
+                role_session_name = validate_role_session_name(role_session_name)
+                role_arn = parse_iam_credentials_arn(role_arn)
+                assumed_role_information = AWSAssumeRoleInfo(
+                    role_arn=role_arn,
+                    session_duration=session_duration,
+                    external_id=external_id,
+                    mfa_enabled=mfa_enabled,
+                    role_session_name=role_session_name,
+                )
+                assumed_role_credentials = AwsProvider.assume_role(
+                    session,
+                    assumed_role_information,
+                )
+                session = Session(
+                    aws_access_key_id=assumed_role_credentials.aws_access_key_id,
+                    aws_secret_access_key=assumed_role_credentials.aws_secret_access_key,
+                    aws_session_token=assumed_role_credentials.aws_session_token,
+                    region_name=aws_region,
+                    profile_name=profile,
+                )
+
+            sts_client = AwsProvider.create_sts_session(session, aws_region)
+            _ = sts_client.get_caller_identity()
+            # Include the region where the caller_identity has validated the credentials
+            return Connection(
+                is_connected=True,
+            )
+
+        except (ClientError, ProfileNotFound, NoCredentialsError) as credentials_error:
+            logger.error(
+                f"{credentials_error.__class__.__name__}[{credentials_error.__traceback__.tb_lineno}]: {credentials_error}"
+            )
+            if raise_on_exception:
+                raise credentials_error
+
+            return Connection(error=credentials_error)
+
+        except ArgumentTypeError as validation_error:
+            logger.error(
+                f"{validation_error.__class__.__name__}[{validation_error.__traceback__.tb_lineno}]: {validation_error}"
+            )
+            if raise_on_exception:
+                raise validation_error
+
+            return Connection(error=validation_error)
```

Comment:
```suggestion
            return Connection(is_connected=False, error=validation_error)
```
Same here

Reply:
`Connection.is_connected` defaults to `False`.

Reply:
Fact